test/extended/builds/sti_secrets.go

```diff
@@ -0,0 +1,105 @@
+package builds
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	"k8s.io/kubernetes/test/e2e"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+var _ = g.Describe("default: S2I incremental build with push and pull to authenticated registry", func() {
+	defer g.GinkgoRecover()
+	var (
+		templateFixture = exutil.FixturePath("fixtures", "incremental-auth-build.json")
+		oc              = exutil.NewCLI("build-sti-env", exutil.KubeConfigPath())
+	)
+
+	g.JustBeforeEach(func() {
+		g.By("waiting for builder service account")
+		err := exutil.WaitForBuilderAccount(oc.KubeREST().ServiceAccounts(oc.Namespace()))
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.Describe("Building from a template", func() {
+		g.It(fmt.Sprintf("should create a build from %q template and run it", templateFixture), func() {
+			oc.SetOutputDir(exutil.TestContext.OutputDir)
+
+			g.By(fmt.Sprintf("calling oc new-app -f %q", templateFixture))
+			err := oc.Run("new-app").Args("-f", templateFixture).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("starting a test build")
+			buildName, err := oc.Run("start-build").Args("initial-build").Output()
```

Comment:
i copy+pasted most of this test from sti_env.go so i can't claim i put any thought into it.

at the end of the day i think it's mostly a style question..  there are a few places where you might save yourself by using a distinct variable but it seems relatively low risk and saves some typing.  (for that matter declaring a new variable for every error could also result in errors, if you then copy+paste a return statement and accidentally return the wrong error variable, for example)